### PR TITLE
🗑️ Scrap Async Instantiation

### PR DESCRIPTION
**Pull request by JuaniRios**
## What?
- Remove async instantiation

## Why?
- Tried making genesis parallel instantiation work, but after the SDK update it seems it's no longer possible due to features like "std" not being passed when calling the runtime for the genesis raw spec creation.
- It is used only in a few tests, and so it's better to delete it all to avoid maintaining it.

## How?
- Delete the async file
- Modify tests that used it
- Remove the populated chain spec

### CODE REVIEW

**Comment by JuaniRios** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/310?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#310** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/310?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#308** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/308?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#307** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/307?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#306** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/306?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#304** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/304?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#303** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/303?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#302** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/302?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#301** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/301?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#299** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/299?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#300](https://github.com/Polimec/polimec-node/pull/300) <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/300?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#296** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/296?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @JuaniRios and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: integration-tests/src/constants.rs [hunk 1]
```diff
@@ -331,6 +333,45 @@ pub mod politest {
 	const GENESIS_PARACHAIN_BOND_RESERVE_PERCENT: Percent = Percent::from_percent(0);
 	const GENESIS_NUM_SELECTED_CANDIDATES: u32 = 5;
 
+	pub fn set_prices() {
```
Comment (vstam1) [inline integration-tests/src/constants.rs:336]: Could make it a general function that you pass the PolitestNet or PolimecNet to, so we don't have it copy pasted. See how we generate the PolimecOrigin and PolimecRuntime types from PolimecNet to see how we could generalize

File: pallets/funding/src/lib.rs [hunk 2]
```diff
@@ -117,6 +117,8 @@
 // This recursion limit is needed because we have too many benchmarks and benchmarking will fail if
 // we add more without this limit.
 #![cfg_attr(feature = "runtime-benchmarks", recursion_limit = "512")]
+extern crate alloc;
```
Comment (vstam1) [inline pallets/funding/src/lib.rs:120]: Why do we need this crate? And should we import it in the necessary file?

File: scripts/zombienet/native/hrmpless-rococo-statemint-polimec-penpal.toml [hunk 3]
```diff
@@ -25,32 +25,32 @@ chain = "rococo-local"
...
+#
+#[[parachains]]
+#id = 1000
+#chain = "asset-hub-rococo-dev"
+#
+#	[[parachains.collators]]
+#	name = "statemint-dev"
+#	command = "../polkadot-sdk/target/release/polkadot-parachain"
+#    args = ["-lerror", "-lxcm::process_instruction=trace", "-lxcm::execute_xcm_in_credit=trace", "-lxcm::barriers=trace", "-lpallet_funding::hrmp=trace", "-lpallet_funding::test=trace", "--execution=native", ]
+#    ws_port = 7777
```
Comment (vstam1) [inline scripts/zombienet/native/hrmpless-rococo-statemint-polimec-penpal.toml:37]: Should we fix this?

**Review by vstam1 — COMMENTED** [review]
Love every piece of code simplification! I was wondering about the whole RefCell setup. I can vaguely remember that it was needed for the Async feature with different execution environments? I was wondering if we could change this RefCell implementation now, as we do not require the std execution environment anymore? I might be completely mistaken, but just a general question.

File: integration-tests/src/constants.rs (hunk 1 above)
Comment (JuaniRios) [reply integration-tests/src/constants.rs:336]: We will remove the politest runtime soon, so we can keep it as is for now

**Comment by JuaniRios** [comment]
> Love every piece of code simplification! I was wondering about the whole RefCell setup. I can vaguely remember that it was needed for the Async feature with different execution environments? I was wondering if we could change this RefCell implementation now, as we do not require the std execution environment anymore? I might be completely mistaken, but just a general question. 
> 
>`RefCell` is there for tests, since we need to setup the mock externalities manually. We use "std" because is needed in both tests and benchmark tests, but cannot be included in the wasm runtime

**Pull request merged**